File: app/src/main/java/com/luffy/mulmedia/egl/EGLSurfaceHolder.java
package com.luffy.mulmedia.egl;

import android.opengl.EGLContext;
import android.opengl.EGLSurface;

public class EGLSurfaceHolder {
    public static final String TAG = "EGLSurfaceHolder";

    private EGLCore mEGLCore = null;

    private EGLSurface mEGLSurface = null;

    public void init(EGLContext context, int flags,int glVersion) {
        mEGLCore = new EGLCore();
        mEGLCore.init(context, flags,glVersion);
    }

    public void createEGLSurface(Object surface, int width, int height) {
        if (surface != null) {
            mEGLSurface = mEGLCore.createWindowSurface(surface);
        } else {
            mEGLSurface = mEGLCore.createOffScreenSurface(width, height);
        }
    }

    public void swapBuffer() {
        if (mEGLSurface != null) {
            mEGLCore.swapBuffers(mEGLSurface);
        }
    }

    public void makeCurrent() {
        if (mEGLSurface != null) {
            mEGLCore.makeCurrent(mEGLSurface);
        }
    }

    public void destroyEGLSurface() {
        if (mEGLCore != null) {
            mEGLCore.destroySurface(mEGLSurface);
            mEGLSurface = null;
        }
    }

    public void release() {
        if (mEGLCore != null) {
            mEGLCore.release();
        }
    }
}

File: opengles-base/src/main/java/com/luffyxu/opengles/base/egl/EGLSurfaceHolder.kt
package com.luffyxu.opengles.base.egl

import android.opengl.EGLContext
import android.opengl.EGLSurface

class EGLSurfaceHolder {
    private var mEGLCore: EGLCore? = null
    private var mEGLSurface: EGLSurface? = null
    fun init(context: EGLContext?, flags: Int, glVersion: Int) {
        mEGLCore = EGLCore()
        mEGLCore!!.init(context, flags, glVersion)
    }

    fun createEGLSurface(surface: Any?, width: Int, height: Int) {
        mEGLSurface = if (surface != null) {
            mEGLCore!!.createWindowSurface(surface)
        } else {
            mEGLCore!!.createOffScreenSurface(width, height)
        }
    }

    fun swapBuffer() {
        if (mEGLSurface != null) {
            mEGLCore!!.swapBuffers(mEGLSurface)
        }
    }

    fun makeCurrent() {
        if (mEGLSurface != null) {
            mEGLCore!!.makeCurrent(mEGLSurface)
        }
    }

    fun destroyEGLSurface() {
        if (mEGLCore != null) {
            mEGLCore!!.destroySurface(mEGLSurface)
            mEGLSurface = null
        }
    }

    fun release() {
        if (mEGLCore != null) {
            mEGLCore!!.release()
        }
    }

    companion object {
        const val TAG = "EGLSurfaceHolder"
    }
}